packages/apollo-gateway/src/__tests__/integration/cucumber.test.ts

```diff
@@ -0,0 +1,141 @@
+import * as books from '../__fixtures__/schemas/books';
+import * as inventory from '../__fixtures__/schemas/inventory';
+import * as product from '../__fixtures__/schemas/product';
+import * as reviews from '../__fixtures__/schemas/reviews';
+
+import { defineFeature, loadFeature } from 'jest-cucumber';
+import { DocumentNode } from 'graphql';
+import { GraphQLExecutionResult } from 'apollo-server-types';
+import { QueryPlan } from '../..';
+
+import { astSerializer, queryPlanSerializer } from '../../snapshotSerializers';
+import prettyFormat from 'pretty-format';
+
+const providesFeature = loadFeature(
+  './packages/apollo-gateway/src/__tests__/integration/provides.feature',
+);
+const abstractTypesFeature = loadFeature(
+  './packages/apollo-gateway/src/__tests__/integration/abstract-types.feature',
+);
+
+let serviceFixtures: { [key: string]: ServiceDefinitionModule } = {
+  accounts,
+  books,
+  inventory,
+  product,
+  reviews,
+};
+
+const features = [
+  // providesFeature,
+  abstractTypesFeature,
+];
+
+features.forEach((feature) => {
+  defineFeature(feature, (test) => {
+    feature.scenarios.forEach((scenario) => {
+      test(scenario.title, async ({ given, when, then }) => {
+        let query: DocumentNode;
+        let variables: any;
+        let services: ServiceDefinitionModule[] = [];
+        let executionResult: GraphQLExecutionResult & { queryPlan: QueryPlan };
+
+        const executeQuery = async () => {
+          if (!executionResult)
+            executionResult = await execute(services, { query, variables });
+        };
+
+        const givenServices = () => {
+          given(/federated (.*) services/i, (servicesString) => {
+            const addedServices = servicesString
+              .split(',')
+              .map((s: string) => s.trim())
+              .map(
+                (name: string) =>
+                  serviceFixtures[name] as ServiceDefinitionModule,
+              );
+            services = services.concat(addedServices)
+          });
+        };
+
+        const givenCustomService = () => {
+          given(/federated (\w*) service with schema/, (name, typeDefs) => {
+            services.push({ name, typeDefs: gql(typeDefs)});
+          })
+        }
+
+        const whenIExecute = () => {
+          when(/^I execute$/im, (operation) => {
+            query = gql(operation);
+          });
+        };
+
+        const andPassVariables = () => {
+          when(/^pass variables$/im, (rawVariables) => {
+            variables = JSON.parse(rawVariables);
+          });
+        };
+
+        const thenServiceShouldOrNotBeCalled = () => {
+          then(
+            /the (.*) service[s]* should (not )*be called/i,
+            async (servicesString, not) => {
+              const serviceNames = servicesString
+                .split(',')
+                .map((s: string) => s.trim());
+
+              await executeQuery();
+              serviceNames.forEach((name: string) => {
```

Comment:
would expectations like this ever catch issues that aren't caught by the query plan snapshot comparison?